packages/contracts/contracts/SmartEscrowMono.sol

```diff
@@ -266,25 +264,36 @@ contract SmartEscrowMono is Context, IArbitrable, ReentrancyGuard {
     override
     nonReentrant
   {
+    // called by aragon court
     require(_ruling <= DISPUTES_POSSIBLE_OUTCOMES, "invalid ruling");
     uint256 index = disputes[_disputeId];
     Locker storage locker = lockers[index];
     require(locker.resolverType == ADR.ARAGON_COURT, "!aragon");
     require(locker.locked, "!locked");
     require(locker.total > locker.released, "released");
     require(_msgSender() == locker.resolver, "!resolver");
-    require(_msgSender() != locker.client, "resolver == client");
 
     uint256 remainder = locker.total.sub(locker.released);
 
-    if (_ruling == DISPUTES_RULING_CLIENT) {
+    if (_ruling == DISPUTES_RULING_100_0) {
       IERC20(locker.token).safeTransfer(locker.client, remainder);
-    } else if (_ruling == DISPUTES_RULING_PROVIDERS) {
+    } else if (_ruling == DISPUTES_RULING_0_100) {
```

Comment:
These else ifs feel like they could get hairy if we want to add more granularity. One potential way to make it cleaner could be to set those rulings as uint8[][] array of arrays, i.e.:

```
rulings = [ 
   [,] , // missing
   [,] , // leaked
   [,] , // refused
   [1,0] , // 100% to client
   [3,1] , // 75% to client
   [1,1] , // 50% to client
   [1,3] , // 25% to client
   [0,1] // 0% to client
];
```
and then access those values with `rulings[_ruling]` with something like this:

```
uint8[] ruling = rulings[_ruling];
uint8 clientShare = ruling[0];
uint8 providerShare = ruling[1];
uint8 denom = clientShare.add(providerShare);
uint256 providerAward = remainder.mul(providerShare).div(denom);
uint256 clientAward = remainder.sub(providerAward);
```

What do you think?

Reply:
Looks great! I'll try this out.